src/pages/patientView/timeline2/VAFChartWrapper.tsx

```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
-                                sampleMetaData.color[sampleId.value] ||
-                                '#333333';
-                            const label =
-                                sampleMetaData.label[sampleId.value] || '-';
-
-                            return (
-                                <g transform={`translate(${x - 7.5},0)`}>
-                                    <SampleMarker color={color} label={label} />
-                                </g>
-                            );
-                        })}
+    @computed get renderData() {
+        return computeRenderData(
+            this.props.samples,
+            this.props.mutations,
+            this.props.sampleIdIndex,
+            this.props.mutationProfileId,
+            this.props.coverageInformation,
+            this.props.store.groupByOption!,
+            this.sampleGroupByValue
+        );
+    }
+
+    @computed get lineData() {
+        let scaledData: IPoint[][] = [];
+        const xposition = this.props.store.getXPositionPerSample();
+        this.renderData.lineData.map((dataPoints: IPoint[], index: number) => {
+            scaledData[index] = [];
+            dataPoints.map((dataPoint: IPoint, i: number) => {
+                scaledData[index].push({
+                    x: xposition[dataPoint.sampleId],
+                    y: this.dataHeight - dataPoint.y * this.dataHeight,
+                    sampleId: dataPoint.sampleId,
+                    mutation: dataPoint.mutation,
+                    mutationStatus: dataPoint.mutationStatus,
+                });
+            });
+        });
+        return scaledData;
+    }
+
+    @computed get mutationToDataPoints() {
+        const map = new ComplexKeyMap<IPoint[]>();
+        for (const lineData of this.lineData) {
+            map.set(
+                {
+                    hugoGeneSymbol: lineData[0].mutation.gene.hugoGeneSymbol,
+                    proteinChange: lineData[0].mutation.proteinChange,
+                },
+                lineData
+            );
+        }
+        return map;
+    }
+
+    @computed get xposition() {
+        return this.props.store.getXPositionPerSample();
+    }
+
+    @computed get sampleIdOrder() {
+        return stringListToIndexSet(
+            this.props.sampleManager.getSampleIdsInOrder()
+        );
+    }
+
+    @computed get sampleGroupByColors() {
+        let groupByColors: { [s: string]: string } = {};
+        const uniqueColorGetter = makeUniqueColorGetter();
+        const clinicalAttributeSamplesMap = this.props.sampleManager.getClinicalAttributeSampleList(
+            this.props.sampleManager.samples,
+            this.props.store.groupByOption!
+        );
+        clinicalAttributeSamplesMap.forEach(
+            (sampleList: string[], clinicalValue: any) => {
+                groupByColors[clinicalValue] = uniqueColorGetter();
+            }
+        );
+        return groupByColors;
+    }
+
+    @computed get sampleGroupByLabels() {
```

Comment:
could you please explain what this is?

Reply:
When user selects a clinical attribute from Group By dropdown, this returns the list of clinical attribute values selected. I use this as labels on the chart.

Reply:
ah okay, I think I understand your phrasing now - its "labels for the sample-group-by feature". I think that we can just go with `clinicalValuesForGrouping`

Reply:
Fixed (changed the name)